### ISSUE TITLE
Participant pair can not be selected

### ISSUE THREAD

**Issue by arnasledev** [livekit/livekit#2589]
**Describe the bug**
While trying to establish a connection between two participants, one failed—and it always fails when this client tries to connect to a video session when other participants are already connected.

If this client connects first, then everything seems to be fine all the time. I assume the bug occurs only when other clients are connected.

**Server**
- Version: v1.5.2
- Environment: Self-managed Kubernetes v1.28.4
- 2x Livekit-server instances with Redis Sentinel

**Client**
- SDK: js/ts
- Version: 2.0.3

**To Reproduce**
1. two clients are connected to room
2. 3rd client joins
3. 3rd client does '...'

**Expected behavior**
Clients should be connected all the time

**Logs**
[console-export.txt](https://github.com/livekit/livekit/files/14666695/console-export.txt)
Logs from our Loki - livekit server logs with this `roomID` filtered [Explore-logs-2024-03-20 14_03_02.json](https://github.com/livekit/livekit/files/14666696/Explore-logs-2024-03-20.14_03_02.json)

**Additional context**
Clients who faced the issue used:
1st client: edge with windows (don't know anything else)
2nd client: Firefox Version 123.0 (64-bit) on macOS Sonoma Version 14.2.1 (23C71)

**Comment by davidzhao** [comment]
@arnasledev are you able to reproduce with https://meet.livekit.io ?

**Comment by arnasledev** [comment]
Yes and no simultaneously - we tried the same and got the same result; I saw participant B as connected, but without the tracks, and after some time (like the 30s), he came alive. Adding logs from the Participant B's perspective

[console-export-2024-4-2_5-23-9.txt](https://github.com/livekit/livekit/files/14833718/console-export-2024-4-2_5-23-9.txt)
